File: cloud/org.kevoree.library.cloud.docker/src/main/java/org/kevoree/library/cloud/docker/wrapper/DockerWrapperFactory.java
package org.kevoree.library.cloud.docker.wrapper;

import org.kevoree.ContainerNode;
import org.kevoree.Instance;
import org.kevoree.api.BootstrapService;
import org.kevoree.api.ModelService;
import org.kevoree.library.DockerNode;
import org.kevoree.library.java.wrapper.KInstanceWrapper;
import org.kevoree.library.java.wrapper.WrapperFactory;
import org.kevoree.pmodeling.api.KMFContainer;

/**
 * Created by duke on 9/29/14.
 */
public class DockerWrapperFactory extends WrapperFactory {

    private DockerNode dockerNode;

    public DockerWrapperFactory(String nodeName,DockerNode dockerNode) {
        super(nodeName);
        this.dockerNode = dockerNode;
    }

    @Override
    public KInstanceWrapper wrap(KMFContainer modelElement, Object newBeanInstance, ThreadGroup tg, BootstrapService bs, ModelService modelService) throws Exception {
        if(modelElement instanceof ContainerNode){
            DockerNodeWrapper w = new DockerNodeWrapper(dockerNode);
            w.setTg(tg);
            w.setBs(bs);
            w.setModelService(modelService);
            w.setTargetObj(newBeanInstance);
            w.setModelElement((Instance) modelElement);
            return w;
        } else {
            return super.wrap(modelElement, newBeanInstance, tg, bs, modelService);
        }
    }
}

File: cloud/org.kevoree.library.cloud.docker/src/main/java/org/kevoree/library/cloud/docker/wrapper/DockerWrapperFactory.kt
package org.kevoree.library.cloud.docker.wrapper

import org.kevoree.modeling.api.KMFContainer
import org.kevoree.api.BootstrapService
import org.kevoree.ContainerNode
import org.kevoree.api.ModelService
import org.kevoree.library.cloud.docker.DockerNode
import org.kevoree.library.java.wrapper.WrapperFactory
import org.kevoree.library.java.wrapper.KInstanceWrapper

/**
 * Created with IntelliJ IDEA.
 * User: leiko
 * Date: 21/05/2014
 * Time: 16:25
 */
class DockerWrapperFactory(nodeName: String, val dockerNode: DockerNode) : WrapperFactory(nodeName) {

    override fun wrap(modelElement: KMFContainer, newBeanInstance: Any, tg: ThreadGroup, bs: BootstrapService, modelService: ModelService): KInstanceWrapper {
        when (modelElement) {
            is ContainerNode -> {
                return DockerNodeWrapper(modelElement, newBeanInstance, tg, bs, dockerNode)
            } else -> {
                return super.wrap(modelElement, newBeanInstance, tg, bs, modelService)
            }
        }
    }
}